asmcomp/mach.mli

```diff
@@ -47,7 +51,6 @@ type operation =
   | Iconst_int of nativeint
   | Iconst_float of int64
   | Iconst_symbol of string
-  | Iconst_blockheader of nativeint
```

Comment:
What was the rationale to introduce `Iconst_blockheader` at the Mach level (in commit 583bfd46c21f66ef6c8b82578289c74f983afd51)?  I understand why it is removed for Spacetime, but I'd like to understand if this means another good property is lost.